lib/widgets/article/article-feed.dart

```diff
@@ -0,0 +1,314 @@
+                },
+              );
+            } else if (snapshot.hasError) {
+              return Text('${snapshot.error}');
+            }
+            return Center(child: const CircularProgressIndicator());
+          }),
+    );
+  }
+
+  AppBar searchBar() {
+    return AppBar(
+        title: Text(
+          'NEWSFEED',
+          textAlign: TextAlign.center,
+        ),
+        centerTitle: true,
+        backgroundColor: Color(0xFF0a0a23));
+  }
+
+  Future<Article> fetchArticle() async {
+    page++;
+
+    await dotenv.load(fileName: ".env");
+
+    String feedUrl =
+        "https://www.freecodecamp.org/news/ghost/api/v3/content/posts/?key=${dotenv.env['NEWSKEY']}&include=tags,authors&page=" +
+            page.toString() +
+            "&fields=title,url,feature_image,id";
+
+    dev.log(feedUrl);
+    final response = await http.get(Uri.parse(feedUrl));
+    if (response.statusCode == 200) {
+      var newArticles = json.decode(response.body)['posts'];
+      setState(() {
+        articles.addAll(newArticles);
+      });
+      return Article.fromJson(jsonDecode(response.body));
+    } else {
+      throw Exception('Failed to load articles');
+    }
+  }
+}
+
+class ArticleTemplate extends StatelessWidget {
+  const ArticleTemplate({
+    Key? key,
+    required this.articels,
+    required this.i,
+  }) : super(key: key);
+
+  final List? articels;
+  final int i;
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+        onTap: () {
+          Navigator.push(
+              context,
+              MaterialPageRoute(
+                  builder: (context) =>
+                      ArticleViewTemplate(articleId: articels?[i]["id"])));
+        },
+        child: Column(
+          children: [
+            ArticleBanner(articels: articels, i: i),
+            Row(
+              children: [
+                Expanded(
+                    child: ConstrainedBox(
+                  constraints: BoxConstraints(minHeight: 50),
+                  child: Container(
+                    color: Color(0xFF0a0a23),
+                    child: Padding(
+                        padding: const EdgeInsets.all(16.0),
+                        child: new InkWell(
+                            child: new Text(truncateStr(articels?[i]["title"]),
+                                style: TextStyle(
+                                    fontSize: 24, color: Colors.white)),
+                            onTap: () => {
+                                  Navigator.push(
+                                      context,
+                                      MaterialPageRoute(
+                                          builder: (context) =>
+                                              ArticleViewTemplate(
+                                                  articleId: articels?[i]
+                                                      ["id"])))
+                                })),
```

Comment:
Same thing here also. You can remove the onTap listeners for the children as the parent has the listener.

```suggestion
                            )),
```